transport.go

```diff
@@ -276,9 +280,13 @@ func (t *HTTPSyncTransport) Configure(options ClientOptions) {
 		}
 	}
 
-	t.client = &http.Client{
-		Transport: t.transport,
-		Timeout:   t.Timeout,
+	if options.HTTPClient != nil {
```

Comment:
I'd suggest to try to refactor the two `Configure` functions somehow.
If I understood the async one has a buffer creation and the `start` call but the rest is the same, so maybe the two public methods could call a private one that prepares things and the async sets the buffer after the return and starts it or else

Reply:
Discussed the issue in person